growattServer/__init__.py

```diff
@@ -580,33 +583,83 @@ def update_plant_settings(self, plant_id, changed_settings, current_settings = N
+        JSON response from the server whether the configuration was successful
         """
-        setting_parameters = parameters
-
+        settings_parameters = parameters
+        
         #If we've been passed an array then convert it into a dictionary
         if isinstance(parameters, list):
-            setting_parameters = {}
+            settings_parameters = {}
             for index, param in enumerate(parameters, start=1):
-                setting_parameters['param' + str(index)] = param
+                settings_parameters['param' + str(index)] = param
+        
+        settings_parameters = {**default_parameters, **settings_parameters}
+
+        response = self.session.post(self.get_url(endpoint), 
+                                     params=settings_parameters)
+        data = json.loads(response.content.decode('utf-8'))
+        return data
 
-        default_params = {
+    def update_mix_inverter_setting(self, serial_number, setting_type, parameters):
+        """
+        Alias for setting inverter parameters on a mix inverter
+        See README for known working settings
+
+        Arguments:
+        serial_number -- Serial number (device_sn) of the inverter (str)
+        setting_type -- Setting to be configured (str)
+        parameters -- Parameters to be sent to the system (dict or list of str)
+                (array which will be converted to a dictionary)
+
+        Returns:
+        JSON response from the server whether the configuration was successful
+        """
+        default_parameters = {
             'op': 'mixSetApiNew',
             'serialNum': serial_number,
             'type': setting_type
         }
-        settings_params = {**default_params, **setting_parameters}
-        response = self.session.post(self.get_url('newTcpsetAPI.do'), params=settings_params)
-        data = json.loads(response.content.decode('utf-8'))
-        return data
+        endpoint = 'newTcpsetAPI.do'
+        return self.update_inverter_setting(serial_number, setting_type, 
+                                            default_parameters, parameters, 
+                                            endpoint)
+
+    def update_ac_inverter_setting(self, serial_number, setting_type, parameters):
+        """
+        Alias for setting inverter parameters on an AC-coupled inverter
+        See README for known working settings
+
+        Arguments:
+        serial_number -- Serial number (device_sn) of the inverter (str)
+        setting_type -- Setting to be configured (str)
+        parameters -- Parameters to be sent to the system (dict or list of str)
+                (array which will be converted to a dictionary)
+
+        Returns:
+        JSON response from the server whether the configuration was successful
+        """
+        default_parameters = {
+            'action': 'spaSet',
+            'serialNum': serial_number,
+            'type': setting_type
+        }
+        endpoint = 'tcpSet.do'
```

Comment:
I'm wondering if `newTcpsetAPI.do` works as endpoint for these as well.
My guess about their APIs is that the API endpoint for changing the settings is the same for every device. However the `op` or `action` is what determines the type of device can be set.

Would you be able to give that a test for me?
setting
```suggestion
        default_parameters = {
            'op': 'spaSetApiNew',
            'serialNum': serial_number,
            'type': setting_type
        }
        endpoint = 'newTcpsetAPI.do'
```

Would be awesome if it works since that means we can use a unified endpoint and don't need to pass that either.
Looking good though! 🙂

Reply:
I thought for certain that this suggestion would work, but it doesn't. tried all the permutations on `op`, `spaSet`|`spaSetApiNew`, and `endpoint` without any joy.

I'll see if I can get an old Android phone and use the method that you and @muppet3000 use. I've not had any success jailbreaking an older iPhone.

Reply:
Ahhh that's a shame, would've been nice to have everything as DRY as we could get it but i guess that this is it for now.

I'll merge this when i have the time available to, i'm assuming later today 🚀 
Also @muppet3000 Thank you so much for the review 💪

Reply:
@indykoning @muppet3000 Hold off on the merge. I've got my Android emulator running and can make this work with the same endpoint. It's a variation on [Indy's suggestion](https://github.com/indykoning/PyPi_GrowattServer/pull/35#discussion_r995758965) with `op=spaSetApi`. I'll put this back to draft until I update the code.

Reply:
OK, I've removed the endpoint now that we have a single one that works for both. Now that I'm using an Android emulator and the SSL packet capture, I've reverted the 'settings discovery' text in the `README.md` file.